examples/api/lib/widgets/navigator_pop_handler/navigator_pop_handler.0.dart

```diff
@@ -71,36 +96,45 @@ class _NestedNavigatorsPageState extends State<NestedNavigatorsPage> {
       },
       child: Navigator(
         key: _nestedNavigatorKey,
+        restorationScopeId: 'nested-navigator',
         initialRoute: 'nested_navigators/one',
         onGenerateRoute: (RouteSettings settings) {
-          switch (settings.name) {
-            case 'nested_navigators/one':
-              final BuildContext rootContext = context;
-              return MaterialPageRoute<void>(
-                builder: (BuildContext context) => NestedNavigatorsPageOne(
-                  onBack: () {
-                    Navigator.of(rootContext).pop();
-                  },
-                ),
-              );
-            case 'nested_navigators/one/another_one':
-              return MaterialPageRoute<void>(
-                builder: (BuildContext context) => const NestedNavigatorsPageTwo(
-                ),
-              );
-            default:
-              throw Exception('Invalid route: ${settings.name}');
-          }
+          final BuildContext rootContext = context;
+          return switch (settings.name) {
+            'nested_navigators/one' => MaterialPageRoute<void>(
+              settings: const RouteSettings(
+                name: 'nested_navigators/one',
+              ),
+              builder: (BuildContext context) => _NestedNavigatorsPageOne(
+                onBack: () {
+                  Navigator.of(rootContext).pop();
+                },
+              ),
+            ),
+            'nested_navigators/one/another_one' => MaterialPageRoute<void>(
+              settings: const RouteSettings(
+                name: 'nested_navigators/one',
+              ),
+              builder: (BuildContext context) => const _NestedNavigatorsPageTwo(),
+            ),
+            _ => MaterialPageRoute<void>(
```

Comment:
is this needed?

Reply:
I could use `_` for the home route... But I think users should probably provide a 404 route in their own apps, and since I hope they copy this example, I think I should provide one too, even though it's not really used here.